src/main/groovy/edu/oregonstate/mist/sadapi/resources/SadResource.groovy

```diff
@@ -0,0 +1,35 @@
+package edu.oregonstate.mist.sadapi.resources
+
+import edu.oregonstate.mist.sadapi.db.SadDAO
+import javax.ws.rs.GET
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.WebApplicationException
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+
+@Path('/')
+class SadResource {
+    private SadDAO sadDAO
+
+    public SadResource(SadDAO sadDAO) {
+        this.sadDAO = sadDAO
+    }
+
+    @GET
+    @Path('{pidm: \\d+}')
+    @Produces(MediaType.APPLICATION_JSON)
+    public Object read(@PathParam('pidm') Long pidm,
+                       @QueryParam('termCodeEntry') String termCodeEntry,
+                       @QueryParam('applNo') Long applNo,
+                       @QueryParam('seqNo') Long seqNo) {
+        if (termCodeEntry && applNo && seqNo)
```

Comment:
@baldrugger in this case it is necessary to test them collectively. if all three query parameters are given in the url, then `queryOne` is called; if none of them are given, `queryAll` is called; otherwise, a bad request response is given.